### DISCUSSION TITLE
Custom client when deployed through Intune doesn't name the computer

### DISCUSSION THREAD

**Discussion by bradin-mcclelland** [rustdesk/rustdesk-server-pro#412] [Q&A] [answered]
So I am trying to deploy rust desk through Intune however the names are only the ID number and not the computer names. Is there any way to change this?
![image](https://github.com/user-attachments/assets/47023cc7-010d-4a41-be55-0ed68688cd88)

**Comment by fufesou** [comment]
> however the names are only the ID number and not the computer names

Hi, I cannot understand the names from the screenshot above.
Can you please give another screenshot of what names you mean?

**Comment by bradin-mcclelland** [comment]
So the first one is a manual install where it has the hostname of the computer and a windows logo and the second one is when its deployed through Intune and doesn't have anything but the ID 

![image](https://github.com/user-attachments/assets/aa9f3f05-5be2-4b25-bb7d-e051eba2fa00)

**Reply by fufesou** [reply]
Thanks.
Are the two peers on the same machine?

**Reply by bradin-mcclelland** [reply]
No they are different machines

**Reply by fufesou** [reply]
This is a weird issue, I've never come across it before.

There may be other reasons that prevent RustDesk from obtaining the user name and machine name.

There is no difference between different installation methods in obtaining the user name and machine name.

Can you please confirm it by manually install RustDesk on the second machine?

**Reply by bradin-mcclelland** [reply]
So it dosent get the name there either but rustdesk does see the name because it comes up when i try to login 
<img width="234" alt="image" src="https://github.com/user-attachments/assets/168533c8-3630-4e16-8610-776699f3c892">

**Reply by bradin-mcclelland** [reply]
The custom client that i use dosent get the name. I know i can manually name them but we have almost a hundred devices and it would take way to long to manually change all of the names on all of the computers

**Reply by fufesou** [reply]
> it would take way to long to manually change all of the names on all of the computers

Yes, this is unacceptable.

But there is no difference between different installation methods in obtaining the user name and machine name.
Could you please try another machine? And try manually installed version on your second machine?

At the same time, I will also look for the reason why RustDesk can't get the username and machine name.

**Reply by fufesou** [reply]
Could you please also give a screenshot of the peer info on the client(local) machine, in file `%appdata%\rustdesk\config\peers\195891537.toml`

![image](https://github.com/user-attachments/assets/f193dedd-3463-44c6-bad6-eb74b3da4951)

**Reply by bradin-mcclelland** [reply]
Here
![image](https://github.com/user-attachments/assets/282933c3-683b-4d94-9a65-cc9d8c57fb5a)

**Reply by fufesou** [reply]
It's the peer info of this one?

![image](https://github.com/user-attachments/assets/d9405b48-1b00-4124-931d-ff10ae5783a9)


So RustDesk can get `username` and `platform`, but the client cannot diplay them for some reasons?

**Reply by bradin-mcclelland** [reply]
Yup. that's what is confusing. its only happening on ones where we use the custom client. if we manually download and add our info manually it shows up just fine

**Reply by fufesou** [reply]
> add our info manually

How do you add the `machine` name and `platform`?

**Reply by bradin-mcclelland** [reply]
Not that info the api key and server and stuff it automatically gets the other info

**Comment by fufesou** [comment]
Can you please delete the `password` in the toml file, and any other privacy peroperties, then upload the toml file here?

![image](https://github.com/user-attachments/assets/3333314f-0c43-4845-8165-1720c52b2e7f)

**Reply by bradin-mcclelland** [reply]
[195891537.toml.txt](https://github.com/user-attachments/files/17330621/195891537.toml.txt)

**Reply by fufesou** [reply]
Well, I am a bit confused.

Peer `195891537` cannot display the username, machine name and system on your client.
But I can see these informations in the configuration file. And they can also be displayed correctly on my client.

The file you uploaded is the one after you have manually installed RustDesk?

**Reply by bradin-mcclelland** [reply]
Yup. i uninstalled and reinstalled the rustdesk. i did use a custom client but thats the only thing i did different between the two

**Reply by fufesou** [reply]
Thanks.

Could you please
1. Uninstall the manullay installed version
2. Delete `%appdata%\rustdesk\config\peers\195891537.toml` on your local machine
3. Install a custom client again
4. Connect the newly installed custom client
5. Find the newly generated toml file on your local machine, then give a screenshot of the peer info or upload the file(after deleting all privacy properties)

**Reply by bradin-mcclelland** [reply]
So i couldnt use the computer i showed you as it was in use but i did have one that was a manual install It had the name and everything i deleted its file and installed the custom client and it shows up fine
![image](https://github.com/user-attachments/assets/4280d76f-0846-4958-a832-226f145374e2)

**Reply by fufesou** [reply]
> but i did have one that was a manual install It had the name and everything i deleted its file and installed the custom client and it shows up fine

No idea why custom client does not work fine on your second machine.

I've tried the custom client, msi package, same configurations as your screenshot.
I can get the username, machine name and platform.

> So i couldnt use the computer i showed you as it was in use

Could you please try again when it is free? Thanks.

**Reply by bradin-mcclelland** [reply]
Could it because it was deployed through intune?

**Reply by fufesou** [reply]
I don't know. But I still think the properties are not related to the ways of installation.

**Comment by bradin-mcclelland** [comment]
So i installed a custom client on a new computer and nothing showed up. so i uninstalled it and installed the client directly from github and the name came up

**Reply by fufesou** [reply]
> So i installed a custom client on a new computer and nothing showed up

Please give a screenshot of the peer info in the toml file. Because the toml file should be invalid if the `platform` is empty. It's wired that your peer has no platform.

![image](https://github.com/user-attachments/assets/bb817e16-5ba2-4b68-a683-80693ef26c8b)

**Reply by bradin-mcclelland** [reply]
This is one that i installed brand new

![image](https://github.com/user-attachments/assets/1d8e37ed-355d-4faf-a621-508cc045a760)

**Reply by fufesou** [reply]
> This is one that i installed brand new

So RustDesk can get the peer info, but the client cannot show the `username`, `hostname` and `platform`?

Could you please 
1. Reopen your RustDesk client
2. Upload the toml file of the newly installed custom client peer, without privacy properties

**Comment by fufesou** [comment]
@bradin-mcclelland Hi, how is issue going? I saw there're infos in the newly installed toml. Is it custom client version and no machine info in the client peer card?

Could you please
1. Reopen your RustDesk client
2. Upload the toml file of the newly installed custom client peer, without privacy properties

**Reply by bradin-mcclelland** [reply]
HI here is the file
[client-77939181.toml.txt](https://github.com/user-attachments/files/17378501/client-77939181.toml.txt)

**Reply by fufesou** [reply]
The properties can be shown correctly in my client after putting the file in `%appdata%\rustdesk\config\peers`.

![image](https://github.com/user-attachments/assets/caac7657-6861-4d14-98cd-b31f849b6b71)


Can't you see this peer's properties in your client?

**Reply by bradin-mcclelland** [reply]
I can not 
![image](https://github.com/user-attachments/assets/bcc1c0ff-339b-4d1b-853c-192095ff8d7c)

**Reply by fufesou** [reply]
So this issue is in the client. Could you please give a screenshot of the version?

![image](https://github.com/user-attachments/assets/698e269e-7a3f-457a-a177-36ca144d79fe)

**Reply by bradin-mcclelland** [reply]
![image](https://github.com/user-attachments/assets/197a2ef4-8067-441d-adaf-98fb9d8fe3d9)

**Reply by fufesou** [reply]
I've downloaded 1.3.1, same build date. But my client can show the properties. 😢

**Reply by fufesou** [reply]
Is the latest nightly build ok for you?

**Reply by bradin-mcclelland** [reply]
nope still dosen't show the name after I updated. I even restarted the computer

**Reply by fufesou** [reply]
Ok.
I'll build a temporary build with some additional logs to locate this issue.
Please help test the temporary build then.
I have to go to bed now.

**Reply by bradin-mcclelland** [reply]
No Problem let me know what i need to do

**Reply by fufesou** [reply]
Please download the temporary  build https://github.com/fufesou/rustdesk/releases/tmp2


1. Make sure `77939181.toml` is still in `%appdata%\rustdesk\config\peers`.
1. Run or install the exe on the local side, then close the client.
2. Open the command line on Windows. `Win` + `R` -> input `cmd` -> Enter

For the installed version
```
"C:\Program Files\RustDesk"\rustdesk.exe | more
```

For the non-installed version
```
%localappdata%\rustdesk\rustdesk.exe | more
```

Please paste the output prefixed with `test load peers =====================`.

**Reply by bradin-mcclelland** [reply]
This what you want?

test load peers =====================
Graphics adapter: Intel(R) Iris(R) Xe Graphics
flutter: launch args: []
flutter: initializing FFI main
flutter: _appType:main,info1-id:TMLT-9MG6R73.TradesmanManufacturing.local,info2-name:TMLT-9MG6R73.TradesmanManufacturing.local,dir:C:\Users\bradin.mcclelland\OneDrive - Tradesman Manufacturing\Documents
flutter: _globalFFI init
flutter: registerEventHandler callback_query_onlines address book peer
flutter: registerEventHandler load_address_book_peers address book peer
flutter: registerEventHandler callback_query_onlines group peer
flutter: registerEventHandler load_group_peers group peer
flutter: registerEventHandler callback_query_onlines recent peer
flutter: registerEventHandler load_recent_peers recent peer
flutter: registerEventHandler callback_query_onlines fav peer
flutter: registerEventHandler load_fav_peers fav peer
flutter: registerEventHandler callback_query_onlines discovered peer
flutter: registerEventHandler load_lan_peers discovered peer
flutter: _globalFFI init end
flutter: registerEventHandler native_ui native_ui
flutter: reset ab model
flutter: registerEventHandler check_software_update_finish check_software_update_finish
flutter: restore lpos: 800.0/600.0, offset:1204.0/411.3333333333333, devicePixelRatio:1.5, isMaximized: false, isFullscreen: false
flutter: restore new offset: 1204.0/411.3333333333333, scale:1.0
flutter: initialLink:
flutter: handled by uni links: false
flutter: [MultiWindowHandler] active window changed: {0}
flutter: pullAb, force: ForcePullAb.listAndCurrent, quiet: false
flutter: pull ab list
flutter: pull ab "My address book"
flutter: pull ab "My address book"

**Reply by fufesou** [reply]
No. `test load peers =====================` is only the prefix.
Please paste full line prefixed with `test load peers =====================`.

No need to paster other lines.

**Reply by bradin-mcclelland** [reply]
ah ok Here it is 
![image](https://github.com/user-attachments/assets/9591459f-3f34-44d4-9c5f-0ad1311890a5)

test load peers ======================= load peers
test load peers ======================= empty username 77939181
test load peers ======================= 77939181, Windows, tmdt-mj03layz,

**Reply by fufesou** [reply]
Also paste lines that prefixed with `flutter: test load peers =====================`, thanks.

**Reply by bradin-mcclelland** [reply]
flutter: test load peers ======================= load, empty username 77939181, original username:
flutter: test load peers ======================= load, 77939181, Windows, tmdt-mj03layz,

**Reply by fufesou** [reply]
No output like `flutter: test load peers ======================= build card, 77939181`?

Are the `hostname` and `platform` of 77939181 still not displayed?

**Reply by bradin-mcclelland** [reply]
I dont see anything like that and it still dosent show the hostname or platform

**Reply by fufesou** [reply]
Thanks a lot. I'll add more logs.

**Reply by fufesou** [reply]
![image](https://github.com/user-attachments/assets/fd05aa09-db36-499b-aac4-08b0f542e3bd)

Can you please check again? If there's `-- More  --`, you should press `Enter` to see the rest output.

**Reply by bradin-mcclelland** [reply]
Nope Nothing

![image](https://github.com/user-attachments/assets/e618499e-a26e-4c91-9348-3ae767a5ae25)

**Reply by fufesou** [reply]
Could you please try this build again? Thanks  https://github.com/fufesou/rustdesk/releases/tmp2

Nothing is changed. Only logs are added.

**Reply by fufesou** [reply]
@bradin-mcclelland Hi, could you please try again https://github.com/fufesou/rustdesk/releases/tmp2

Sorry for the inconvenience, but I could not reproduce this issue.

The only way is to test with logs.

**Reply by bradin-mcclelland** [reply]
C:\Users\bradin.mcclelland>"C:\Program Files\RustDesk"\rustdesk.exe | more
Graphics adapter: Intel(R) Iris(R) Xe Graphics
flutter: launch args: [\rustdesk.exe]
flutter: initializing FFI main
flutter: _appType:main,info1-id:TMLT-9MG6R73.TradesmanManufacturing.local,info2-name:TMLT-9MG6R73.TradesmanManufacturing.local,dir:C:\Users\bradin.mcclelland\OneDrive - Tradesman Manufacturing\Documents
flutter: _globalFFI init
flutter: registerEventHandler callback_query_onlines address book peer
flutter: registerEventHandler load_address_book_peers address book peer
flutter: registerEventHandler callback_query_onlines group peer
flutter: registerEventHandler load_group_peers group peer
flutter: registerEventHandler callback_query_onlines recent peer
flutter: registerEventHandler load_recent_peers recent peer
flutter: registerEventHandler callback_query_onlines fav peer
flutter: registerEventHandler load_fav_peers fav peer
flutter: registerEventHandler callback_query_onlines discovered peer
flutter: registerEventHandler load_lan_peers discovered peer
flutter: _globalFFI init end
flutter: registerEventHandler native_ui native_ui
flutter: reset ab model
flutter: registerEventHandler check_software_update_finish check_software_update_finish
flutter: test load peers ======================= _buildPeersView, empty username 369241462}
flutter: test load peers ======================= _buildPeersView, empty platform 369241462
flutter: test load peers ======================= _buildPeersView, empty username 77939181}
flutter: test load peers ======================= _buildPeersView, empty hostname 77939181}
flutter: test load peers ======================= _buildPeersView, empty platform 77939181
flutter: test load peers ======================= _buildPeersView, 77939181, , ,
flutter: test load peers ======================= _buildPeersView, empty username 126518851}
flutter: test load peers ======================= _buildPeersView, empty platform 126518851
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 79971634}
flutter: test load peers ======================= _buildPeersView, empty hostname 79971634}
flutter: test load peers ======================= _buildPeersView, empty platform 79971634
flutter: test load peers ======================= _buildPeersView, empty username 369241462}
flutter: test load peers ======================= _buildPeersView, empty platform 369241462
flutter: test load peers ======================= _buildPeersView, empty username 77939181}
flutter: test load peers ======================= _buildPeersView, empty hostname 77939181}
flutter: test load peers ======================= _buildPeersView, empty platform 77939181
flutter: test load peers ======================= _buildPeersView, 77939181, , ,
flutter: test load peers ======================= _buildPeersView, empty username 126518851}
flutter: test load peers ======================= _buildPeersView, empty platform 126518851
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 79971634}
flutter: test load peers ======================= _buildPeersView, empty hostname 79971634}
flutter: test load peers ======================= _buildPeersView, empty platform 79971634
flutter: test load peers ======================= before build, empty username 126518851}
flutter: test load peers ======================= before build, empty platform 126518851
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 369241462}
flutter: test load peers ======================= before build, empty platform 369241462
flutter: test load peers ======================= before build, empty username 77939181}
flutter: test load peers ======================= before build, empty hostname 77939181}
flutter: test load peers ======================= before build, empty platform 77939181
flutter: test load peers ======================= before build, 77939181, , ,
flutter: test load peers ======================= before build, empty username 79971634}
flutter: test load peers ======================= before build, empty hostname 79971634}
flutter: test load peers ======================= before build, empty platform 79971634
flutter: restore lpos: 800.0/600.0, offset:256.6666666666667/0.0, devicePixelRatio:1.5, isMaximized: false, isFullscreen: false
flutter: restore new offset: 256.6666666666667/0.0, scale:1.0
flutter: initialLink: \rustdesk.exe
flutter: handled by uni links: false
flutter: [MultiWindowHandler] active window changed: {0}
flutter: pullAb, force: ForcePullAb.listAndCurrent, quiet: false
flutter: test load peers ======================= _buildPeersView, empty username 369241462}
flutter: test load peers ======================= _buildPeersView, empty platform 369241462
flutter: test load peers ======================= _buildPeersView, empty username 77939181}
flutter: test load peers ======================= _buildPeersView, empty hostname 77939181}
flutter: test load peers ======================= _buildPeersView, empty platform 77939181
flutter: test load peers ======================= _buildPeersView, 77939181, , ,
flutter: test load peers ======================= _buildPeersView, empty username 126518851}
flutter: test load peers ======================= _buildPeersView, empty platform 126518851
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 79971634}
flutter: test load peers ======================= _buildPeersView, empty hostname 79971634}
flutter: test load peers ======================= _buildPeersView, empty platform 79971634
flutter: test load peers ======================= before build, empty username 126518851}
flutter: test load peers ======================= before build, empty platform 126518851
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 369241462}
flutter: test load peers ======================= before build, empty platform 369241462
flutter: test load peers ======================= before build, empty username 77939181}
flutter: test load peers ======================= before build, empty hostname 77939181}
flutter: test load peers ======================= before build, empty platform 77939181
flutter: test load peers ======================= before build, 77939181, , ,
flutter: test load peers ======================= before build, empty username 79971634}
flutter: test load peers ======================= before build, empty hostname 79971634}
flutter: test load peers ======================= before build, empty platform 79971634
flutter: test load peers ======================= before build, empty username 126518851}
flutter: test load peers ======================= before build, empty platform 126518851
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 369241462}
flutter: test load peers ======================= before build, empty platform 369241462
flutter: test load peers ======================= before build, empty username 77939181}
flutter: test load peers ======================= before build, empty hostname 77939181}
flutter: test load peers ======================= before build, empty platform 77939181
flutter: test load peers ======================= before build, 77939181, , ,
flutter: test load peers ======================= before build, empty username 79971634}
flutter: test load peers ======================= before build, empty hostname 79971634}
flutter: test load peers ======================= before build, empty platform 79971634
flutter: pull ab list
flutter: test load peers ======================= _buildPeersView, empty username 369241462}
flutter: test load peers ======================= _buildPeersView, empty platform 369241462
flutter: test load peers ======================= _buildPeersView, empty username 77939181}
flutter: test load peers ======================= _buildPeersView, empty hostname 77939181}
flutter: test load peers ======================= _buildPeersView, empty platform 77939181
flutter: test load peers ======================= _buildPeersView, 77939181, , ,
flutter: test load peers ======================= _buildPeersView, empty username 126518851}
flutter: test load peers ======================= _buildPeersView, empty platform 126518851
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 79971634}
flutter: test load peers ======================= _buildPeersView, empty hostname 79971634}
flutter: test load peers ======================= _buildPeersView, empty platform 79971634
flutter: test load peers ======================= before build, empty username 126518851}
flutter: test load peers ======================= before build, empty platform 126518851
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 369241462}
flutter: test load peers ======================= before build, empty platform 369241462
flutter: test load peers ======================= before build, empty username 77939181}
flutter: test load peers ======================= before build, empty hostname 77939181}
flutter: test load peers ======================= before build, empty platform 77939181
flutter: test load peers ======================= before build, 77939181, , ,
flutter: test load peers ======================= before build, empty username 79971634}
flutter: test load peers ======================= before build, empty hostname 79971634}
flutter: test load peers ======================= before build, empty platform 79971634
flutter: pull ab "My address book"
flutter: pull ab "My address book"
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 369241462}
flutter: test load peers ======================= _buildPeersView, empty platform 369241462
flutter: test load peers ======================= _buildPeersView, empty username 77939181}
flutter: test load peers ======================= _buildPeersView, empty hostname 77939181}
flutter: test load peers ======================= _buildPeersView, empty platform 77939181
flutter: test load peers ======================= _buildPeersView, 77939181, , ,
flutter: test load peers ======================= _buildPeersView, empty username 126518851}
flutter: test load peers ======================= _buildPeersView, empty platform 126518851
flutter: test load peers ======================= _buildPeersView, empty username 195891537}
flutter: test load peers ======================= _buildPeersView, empty hostname 195891537}
flutter: test load peers ======================= _buildPeersView, empty platform 195891537
flutter: test load peers ======================= _buildPeersView, empty username 79971634}
flutter: test load peers ======================= _buildPeersView, empty hostname 79971634}
flutter: test load peers ======================= _buildPeersView, empty platform 79971634
flutter: test load peers ======================= before build, empty username 126518851}
flutter: test load peers ======================= before build, empty platform 126518851
flutter: test load peers ======================= before build, empty username 195891537}
flutter: test load peers ======================= before build, empty hostname 195891537}
flutter: test load peers ======================= before build, empty platform 195891537
flutter: test load peers ======================= before build, empty username 369241462}
flutter: test load peers ======================= before build, empty platform 369241462
flutter: test load peers ======================= before build, empty username 77939181}
flutter: test load peers ======================= before build, empty hostname 77939181}
flutter: test load peers ======================= before build, empty platform 77939181
flutter: test load peers ======================= before build, 77939181, , ,
flutter: test load peers ======================= before build, empty username 79971634}
flutter: test load peers ======================= before build, empty hostname 79971634}
-- More  -- test load peers ======================= load peers
test load peers ======================= empty username 77939181
test load peers ======================= 77939181, Windows, tmdt-mj03layz,

**Comment by fufesou** [comment]
Hi, which tab do you see this empty peer card?

![image](https://github.com/user-attachments/assets/57db6138-b197-4b95-8657-c74ec238fba6)

Is it the group tab? And the recent(left most) tab shows the correct properties?

![image](https://github.com/user-attachments/assets/4f6068d8-e224-4a30-92f6-ca6297d5470b)

Please ensure if the API setting is correct if so.

https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/faq/#why-are-my-logs--device-names-are-empty

https://github.com/rustdesk/rustdesk-server-pro/issues/421#issuecomment-2419517071

**Reply by bradin-mcclelland** [reply]
So for the api key i cant put the port in or use http it causes errors but if i use https and no port number it works fine.

I see the empty ones in both

**Reply by fufesou** [reply]
> I see the empty ones in both

Could you please confirm:

1. Is the Card `77939181` of recent tab empty?
2. Find one empty peer card of recent tab, check if the porperties in `%appdata%\rustdesk\config\peers\<id>.toml` are empty?

I see `77939181`'s properties in the self-hosted server db are emtpy. Maybe the API server in that machine's client is not correct.

Could you please also give a screenshot of the Network settings? Please smudge the image to ensure that no privacy information is exposed, cover the domains and the `Key`.

**Reply by bradin-mcclelland** [reply]
Here is the settings is this what you mean? when i try to put the exact api settings into my client it dosent work and comes up with thiss error message
<img width="302" alt="image" src="https://github.com/user-attachments/assets/d52a50f2-e018-40d7-9e9f-1e5f712c62f3">

<img width="382" alt="image" src="https://github.com/user-attachments/assets/1a94a5a3-2c42-4b67-85b9-a68cbb290116">

**Reply by fufesou** [reply, marked as answer]
You need to change your custom client configuration and rebuild your custom client. `API server` cannot be changed for custom client. So it always is `http://rustdesk.***.ca:21114`

![image](https://github.com/user-attachments/assets/95b1ba0d-8bbe-4c6d-9991-0d956ca9d556)



![image](https://github.com/user-attachments/assets/13147029-8ec9-474d-82fe-cfab2f026c27)

**Reply by bradin-mcclelland** [reply]
That was the issue. im not sure why the api needs to not have the port information but it is working now. thank you for your help

**Comment by PC-Support-Services** [comment]
I had this issue as well and the API server fix resolved it.

Latest pro server version.
